.github/workflows/rule-validate.yml

```diff
@@ -92,6 +93,46 @@ jobs:
         id: get_head
         run: echo "##[set-output name=HEAD;]$(git rev-parse HEAD)"
 
+      - name: Get changed detection-rules
+        id: changed-files
+        uses: tj-actions/changed-files@v39
+        with:
+          files: "detection-rules/**"
+          recover_deleted_files: true
+
+      - name: "Find updated rule IDs"
+        id: find_ids
+        run: |
+          for file in ${{ steps.changed-files.outputs.all_changed_and_modified_files }}; do
+            echo "$file was changed"
+            rule_id=$(yq '.id' $file)
+            [[ "$rule_id" = "null" ]] && echo "$file has no id yet" && continue
+          
+            echo "$file has rule ID $rule_id"
+            altered_rule_ids=$(echo "$rule_id"" ""$altered_rule_ids")
+          done
+          
+          echo "Altered Ruled IDs: [$altered_rule_ids]"
+          echo "##[set-output name=rule_ids;]$(echo $altered_rule_ids)"
+          # TODO: This doesn't solve for a modified rule_id. We could merge with any files known on 'main', but changing
+          # a rule ID is a separate problem.
+
+      - name: "Trigger MQL Mimic Tests"
+        env:
+          trigger_url: '${{ secrets.MQL_MOCK_TRIGGER }}'
+          branch: ${{ github.event_name == 'pull_request_target' && github.head_ref || github.ref }}
```

Comment:
On `main`/etc `github.ref` will include "refs/heads/". I decided to just handle that downstream.